dockerfile_parse/parser.py

```diff
@@ -275,6 +277,13 @@ def _clean_comment_line(line):
             # It is necessary to keep instructions and comment parsing separate,
             # as a multi-line instruction can be interjected with comments.
             if commentre.match(line):
+                # once support for python versions before 3.8 is dropped use walrus operator
+                if lineno == 0 and directivere.match(line):
```

Comment:
If I read docs correctly, `lineno == 0` is not always true. Directive can be on next lines if all previous lines are comments.

Reply:
Ahh sorry, I read it incorrectly

Reply:
I think you are right here, the current implementation will fail if there is first a `syntax` directive and in the next line a `escape` directive. But the docs support this case